src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -104,6 +113,7 @@ public Task<Session> TryCreateServiceSessionAsync(string serviceName, Solution s
         [Obsolete]
         protected virtual Task<Session> CreateServiceSessionAsync(string serviceName, PinnedRemotableDataScope snapshot, object callbackTarget, CancellationToken cancellationToken)
         {
+            snapshot?.Dispose();
```

Comment:
💭 It's possible we could just delete this method altogether and make the caller abstract.